src/Directives.jl

```diff
@@ -95,11 +99,20 @@ function parsebracket end
 parsebracket(text :: AbstractString) = parsebracket(getdirective(text)...)
 parsebracket{D}(:: Directive{D}, text) = error("Unknown directive: '$D'")
 
+function parsebracket{D}(TD :: Directive{D}, arg, text)
+    if strip(arg) == ""
+        parsebracket(TD, text)
+    else
+        error("parsebracket not defined for directive '$D' with positional argument.")
+    end
+end
+
 function parsebracket(:: directive"docs", text)
+    eval(Expr(:toplevel, Expr(:using, :Lexicon)))
```

Comment:
``@object`` will need to be used [here](https://github.com/MichaelHatherly/Docile.jl/blob/e463a72aa071e2b2353ae2d02eda975895087e0d/src/Directives/extensions.jl#L4-L23).

You'd need to add another field to ``Documentation``, something like ``objects :: Vector``, and also construct and pass another ``Expr(:vcat, ...)`` to to constructor in ``directives(:: D"docs", text)`` with entries ``:(@object($(parse(s))))``.

After that all the magic will happen in ``writemime`` where you can generate a signature based on each object, display it and then display the docstring itself. It may still be a little tricky to get the correct methods that are local rather than all of them though.

(If you've seen yesterday's commits then you might realise that I've moved most of Lexicon's features over to Docile. Since we'll have precompilation in 0.4 there's no real need to split the packages anymore. Making changes should be quite a bit less painful with just one codebase to deal with.)

> I have been roped into giving a (brief) talk on this next month

That's cool, I think we may be sorted by then, hopefully. If you need to confirm/check any details them feel free to email me.